x/avs/keeper/impl_epoch_hook.go

```diff
@@ -35,68 +35,59 @@ func (wrapper EpochsHooksWrapper) AfterEpochEnd(
 	// TODO:There should be a retry mechanism or compensation mechanism to handle cases of failure
 	if len(taskResList) != 0 {
 		groupedTasks := wrapper.keeper.GroupTasksByIDAndAddress(taskResList)
-		for _, value := range groupedTasks {
+		for key, value := range groupedTasks {
+			taskAddr, taskID, _ := parseGroupKey(key)
+			avsInfo := wrapper.keeper.GetAVSInfoByTaskAddress(ctx, taskAddr)
```

Comment:
GetAVSInfoByTaskAddress may return empty AVSInfo, there is no validation to check if avsAddr is empty.